types/runtime.d.ts

```diff
@@ -48,7 +48,22 @@ export interface RuntimeProvider {
   // Apply provider base
   // Add additional params (like signuture)
   // Do modifier mapping
-  generateURL: (src: string, modifiers: ImageModifiers, providerOptions: any) => { url: string, isStatic?: boolean }
+  getImage: (src: string, modifiers: ImageModifiers, providerOptions: any) => RuntimeImage
+}
+
+export interface RuntimeImage {
+  url: string,
+  isStatic?: boolean,
+  getInfo?: () => Promise<RuntimeImageInfo>
+}
+
+export interface RuntimeImageInfo {
+  // width of image in pixels
+  width: number,
+  // height of image in pixels
+  height: number,
+  // size of image in bytes
+  size: number,
```

Comment:
What if call it `bytes` to not confuse with px size and also making self explain that unit is bytes?